fe/fe-core/src/main/java/org/apache/doris/external/elasticsearch/QueryBuilders.java

```diff
@@ -38,6 +59,238 @@
  */
 public final class QueryBuilders {
 
+    /**
+     * Generate dsl from compound expr.
+     **/
+    private static QueryBuilder toCompoundEsDsl(Expr expr, List<Expr> notPushDownList,
+            Map<String, String> fieldsContext, boolean likePushdown) {
```

Comment:
should we use a `BuilderOptions` or `BuilderContext` to wrap the `boolean likePushdown` if it has more option later?

Reply:
> should we use a `BuilderOptions` or `BuilderContext` to wrap the `boolean likePushdown` if it has more option later?

Good idea.

Reply:
Okay, I will modify it.